nwastdlib/asyncio_cache.py

```diff
@@ -0,0 +1,169 @@
+#  Copyright 2019 SURF.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#          http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+import hashlib
+import hmac
+import pickle
+from functools import wraps
+from typing import Any, Callable
+
+import structlog
+from redis.asyncio import Redis as AIORedis
+
+logger = structlog.get_logger(__name__)
+
+
+def get_hmac_checksum(secret, message):
+    # Todo: check if the current secret is good enough for sha512, it might be better to use a new SECRET just for this
+    h = hmac.new(secret.encode(), message, hashlib.sha512)
+    return h.hexdigest()
+
+
+async def set_signed_cache_value(pool: AIORedis, secret: str, cache_key: str, value: Any, expiry: int):
```

Comment:
Variable name can indicate the unit
```suggestion
async def set_signed_cache_value(pool: AIORedis, secret: str, cache_key: str, value: Any, expiry_seconds: int):
```

Reply:
Done